mesonbuild/interpreter.py

```diff
@@ -1798,7 +1798,12 @@ def dependency_fallback(self, name, kwargs):
         dirname, varname = self.get_subproject_infos(kwargs)
         # Try to execute the subproject
         try:
-            self.do_subproject(dirname, {})
+            sp_kwargs = {}
+            try:
+                sp_kwargs['default_options'] = kwargs['default_options']
```

Comment:
I tested and this snippet can be run 2 million times a second. I'd say that is fast enough.